index.ts

```diff
@@ -18,6 +18,13 @@ declare namespace ethereum {
 /** Host IPFS interface */
 export declare namespace ipfs {
   function cat(hash: string): Bytes | null
+  function map(hash: string, callback: string, flags: string[]): void
+}
+
+export namespace ipfs {
+    export function mapJSON(hash: string, callback: string): void {
```

Comment:
I see 4 spaces of indentation here and in the next line.

Reply:
My emacs settings are getting the better of me.